runtime/vm/options.c

```diff
@@ -0,0 +1,176 @@
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+
+#include <stdio.h>
+#include "j9protos.h"
+#include "jvminitcommon.h"
+#include "options.h"
+
+
+
+/* Option types for compatibility */
+#define TYPE_BOOL 0
+#define TYPE_CCSTR 1
+#define TYPE_CCSTRLIST 2
+#define TYPE_DOUBLE 3
+#define TYPE_INT 4
+#define TYPE_INTX 5
+#define TYPE_SIZET 6
+#define TYPE_UINT 7
+#define TYPE_UINT64T 8
+#define TYPE_UINTX 9
+
+static char *option_type_string[] = {
+	"bool",
+	"ccstr",
+	"ccstrlist",
+	"double",
+	"int",
+	"intx",
+	"size_t",
+	"uint",
+	"uint64_t",
+	"uintx"
+};
+
+typedef union J9OptionValue {
+	UDATA v_uintx;
+	U_64 v_uint64;
+	size_t v_sizet;
+	double v_double;
+	char *v_ccstr;
+	char **v_ccstrlist;
+	U_32 v_uint;
+	IDATA v_intx;
+	I_8 v_bool;
+} J9OptionValue;
+
+#define CAT_PRODUCT		0
+#define CAT_ARCHPRODUCT		1
+#define CAT_PDPRODUCT		2
+#define CAT_PRODUCTRW		3
+#define CAT_DIAGNOSTIC		4
+#define CAT_EXPERIMENTAL	5
+#define CAT_MANAGEABLE		6
+
+static char *option_category_string[] = {
+	"{product}",
+	"{ARCH product}",
+	"{pd product}",
+	"{product rw}",
+	"{diagnostic}",
+	"{experimental}",
+	"{manageable}",
+};
+
+#define SET_DEFAULT	0
+#define SET_COMMANDLINE	1
+#define SET_ERGONOMIC	2
+
+static char *option_howset_string[] = {
+	"{default}",
+	"{command line}",
+	"{ergonomic}"
+};
+
+typedef struct J9OptionDesc {
+	char * name;
+	I_8 type;
+	I_8 category;
+	I_8 howset;
+	J9OptionValue value;
+} J9OptionDesc;
+
+/*
+ * [Global flags]
+ *    size_t MaxHeapSize                              = 4294967296                                {product} {ergonomic}
+ *   uint64_t MaxDirectMemorySize                     = 3758096384                                         {product} {default}
+ */
+
+
+typedef enum options_enum {
+	NoOption=-1,
+
+	#define OPTION(n,t,cats) IDX_ ## n,
+	#include "options.inc"
+	#undef OPTION
+} options_enum;
+
+static J9OptionDesc options[]={
+	#define OPTION(n,t,cats) { n, t, cats, SET_DEFAULT, { 0 } },
+	#include "options.inc"
+	#undef OPTION
+};
+#define NUM_OPTIONS	(sizeof(options)/sizeof(J9OptionDesc))
+
+
+#define VMOPTION(o) VMOPT_ ## o
+
+#define SET_OPTION(option, cmdLineOption, vtype, setValue, defaultSource) \
+	o = options + option; \
+	o->value.vtype = (setValue); \
+	if ((findArgInVMArgs( PORTLIB, vm->vmArgsArray, STARTSWITH_MATCH, cmdLineOption, NULL, 0)) >= 0) { \
```

Comment:
-XX:MaxHeapSize is an alias for -Xmx. I expect setting -Xmx should indicate that -XX:MaxHeapSize was set on the command line.

Reply:
true, I guess I'll need to incorporate that alias too. thanks!

I've addressed it with the latest update.